### PR TITLE
ffpb: init at 0.4.1

### PR DESCRIPTION
**Pull request by CaptainJawZ**
## Description of changes

Init for ffpb https://github.com/althonos/ffpb

> ffpb is an FFmpeg progress formatter. It will attempt to display a nice progress bar in the output, based on the raw ffmpeg output, as well as an adaptative ETA timer.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/ffpb/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  };
+
+  propagatedBuildInputs = [ tqdm ];
+
+  # tests require working internet connection
+  doCheck = false;
+
+  meta = with lib; {
+    description =
+      "ffpb is an FFmpeg progress formatter. It will attempt to display a nice progress bar in the output, based on the raw ffmpeg output, as well as an adaptative ETA timer.";
```
Comment (FliegendeWurst) [inline pkgs/development/python-modules/ffpb/default.nix:20]:
```suggestion
    description = "FFmpeg progress formatter to display a nice progress bar and an adaptative ETA timer";
```

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -21119,6 +21119,8 @@ with pkgs;
 
   ffms = callPackage ../development/libraries/ffms { };
 
+  ffpb = with python3Packages; toPythonApplication ffpb;
```
[the remark this answers is not in the document]
Comment (FliegendeWurst) [reply pkgs/top-level/all-packages.nix:21122]: And please make sure your files are formatted.

[the remark this answers is not in the document]
Comment (CaptainJawZ) [reply pkgs/top-level/all-packages.nix:21122]: apologies, I made this package with the old formatter, I'll update the commit first thing in the morning with the required changes.

File: pkgs/by-name/ff/ffpb/package.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  propagatedBuildInputs = [ python3Packages.tqdm ];
+
+  # tests require working internet connection
+  doCheck = false;
+
+  meta = {
+    description = "FFmpeg progress formatter to display a nice progress bar and an adaptative ETA timer";
+    homepage = "https://github.com/althonos/ffpb";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ CaptainJawZ ];
```
Comment (FliegendeWurst) [inline pkgs/by-name/ff/ffpb/package.nix:26]:
```suggestion
    maintainers = with lib.maintainers; [ CaptainJawZ ];
    mainProgram = "ffpb";
```

File: pkgs/by-name/ff/ffpb/package.nix [hunk 4]
```diff
@@ -0,0 +1,29 @@
+{
+  lib,
+  python3Packages,
+  fetchPypi,
+}:
+
+python3Packages.buildPythonPackage rec {
```
Comment (FliegendeWurst) [inline pkgs/by-name/ff/ffpb/package.nix:7]:
```suggestion
python3Packages.buildPythonApplication rec {
```
Since it is only used as a binary.

Comment (CaptainJawZ) [reply pkgs/by-name/ff/ffpb/package.nix]: merged the suggestions + squashed into a single commit n_n

**Review by FliegendeWurst — APPROVED** [review]

File: pkgs/by-name/ff/ffpb/package.nix [hunk 5]
```diff
@@ -0,0 +1,29 @@
+{
+  lib,
+  python3Packages,
+  fetchPypi,
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "ffpb";
+  version = "0.4.1";
+  format = "setuptools";
```
Comment (FliegendeWurst) [inline pkgs/by-name/ff/ffpb/package.nix:10]:
```suggestion
  pyproject = true;
```

File: pkgs/by-name/ff/ffpb/package.nix [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "7eVqbLpMHS1sBw2vYS4cTtyVdnnknGtEI8190VlXflk=";
+  };
+
+  propagatedBuildInputs = [ python3Packages.tqdm ];
```
Comment (FliegendeWurst) [inline pkgs/by-name/ff/ffpb/package.nix:17]:
```suggestion
  build-system = [ python3Packages.setuptools ];

  dependencies = [ python3Packages.tqdm ];
```
I recently learned that even packages without pyproject.toml can (and should) use the new pyproject builder.

File: pkgs/by-name/ff/ffpb/package.nix [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+  # tests require working internet connection
+  doCheck = false;
+
+  meta = {
+    description = "FFmpeg progress formatter to display a nice progress bar and an adaptative ETA timer";
+    homepage = "https://github.com/althonos/ffpb";
```
Comment (FliegendeWurst) [inline pkgs/by-name/ff/ffpb/package.nix:24]:
```suggestion
    homepage = "https://github.com/althonos/ffpb";
    changelog = "https://github.com/althonos/ffpb/blob/master/CHANGELOG.md";
```

**Comment by CaptainJawZ** [comment]
Thank you for your feedback and the info @FliegendeWurst 
I've implemented your changes and pushed a new version.

Let me know if there is anything else I could adjust.

**Review by FliegendeWurst — APPROVED** [review]

**Pull request merged**